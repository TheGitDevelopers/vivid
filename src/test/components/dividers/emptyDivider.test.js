import React from 'react';
import renderer from 'react-test-renderer';
import EmptyDivider from '../../../components/common/dividers/emptyDivider';

describe('EmptyDivider tests:', () => {
  /* Test mockups */
  const tree = renderer
    .create(<EmptyDivider />);

  /* All component tests should be listed here */
  it('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('default size prop is medium', () => {
    expect(tree.root.findByType(EmptyDivider).props.size).toBe('medium');
  });
  it('medium size is a 16 marginVertical value', () => {
    const childStyle = tree.root.findByType(EmptyDivider).children[0].props.style;
    const marginVerticalProp = childStyle.filter((e) => (e ? Object.keys(e)[0] === 'marginVertical' : null))[0].marginVertical;
    expect(marginVerticalProp).toBe(16);
  });
});
