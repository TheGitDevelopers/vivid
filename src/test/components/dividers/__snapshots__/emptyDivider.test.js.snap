// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyDivider tests: renders correctly 1`] = `
<View
  style={
    Array [
      null,
      null,
      Object {
        "marginVertical": 16,
      },
      null,
    ]
  }
/>
`;
